gitlab.go

```diff
@@ -50,53 +51,60 @@ func (client *gitLabClient) listProjects(page int) ([]*gitlab.Project, gitLabPag
 }
 
 func (client *gitLabClient) getProject(groupName string, projectName string) (*gitlab.Project, error) {
-	projects, _, err := client.listProjectsFromGroup(fmt.Sprintf("%v/%v", groupName, projectName), 0)
+	project, _, err := client.Projects.GetProject(fmt.Sprintf("%v/%v", groupName, projectName), nil, nil)
 	if err != nil {
-		log.Error().
+		ev := log.Error().
 			WithError(err).
-			WithStringf("project", "%s/%s", groupName, projectName).
-			Message("Failed to list projects for project name and group.")
-		return nil, err
-	}
-
-	var filteredProjects []*gitlab.Project
-	for _, p := range projects {
-		if p.Name == projectName || p.Path == projectName {
-			filteredProjects = append(filteredProjects, p)
+			WithString("group", groupName).
+			WithString("project", projectName)
+		ev.Message("Failed to get project.")
+		projects, _, err := client.Search.Projects(projectName, &gitlab.SearchOptions{})
+		if err == nil {
+			for _, proj := range projects {
+				if strings.EqualFold(proj.Namespace.FullPath, groupName) ||
+					strings.EqualFold(proj.Namespace.Name, groupName) ||
+					strings.EqualFold(proj.Namespace.Path, groupName) {
```

Comment:
Don't think we'd want to check the group display name and last path segment. Was that the behaviour before? I think that could be quite volatile.

I think we pretty much only want to check the full path, given it correlates to the `path_with_namespace` JSON field (https://docs.gitlab.com/ee/api/projects.html#list-all-projects)

```suggestion
				if strings.EqualFold(proj.Namespace.FullPath, groupName) {
```